nautobot/users/tests/test_api.py

```diff
@@ -73,6 +73,69 @@ def setUpTestData(cls):
         Group.objects.create(name="Group 3")
 
 
+class TokenTest(APIViewTestCases.APIViewTestCase):
+    model = Token
+    brief_fields = ["display", "id", "key", "url", "write_enabled"]
+    bulk_update_data = {
+        "description": "New description",
+    }
+
+    def setUp(self):
+        super().setUp()
+
+        tokens = (
+            # We already start with one Token, created by the test class
+            Token(user=self.user),
+            Token(user=self.user),
+        )
+        # Use save() instead of bulk_create() to ensure keys get automatically generated
+        for token in tokens:
+            token.save()
+
+        self.create_data = [
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+        ]
```

Comment:
So this is interesting... reading this here made me realize that `TokenViewSet` is actually a full-featured API view set of its own, meaning that it can itself be used to create, bulk-create, update, and bulk-update Tokens, in addition to the specific Token creation flow enabled by `TokenProvisionView`. The overridden `get_queryset` method does mean that a non-superuser can at least only create/update tokens for their own user account, so I think this is OK, but just leaving a comment here as I'd missed this on initial review.

Reply:
On re-review I actually have some uncertainty here - would you mind adding a few more API test cases for safety:

- User A tries to `create` a token belonging to User B (should fail with an appropriate error)
- User B queries for a list of tokens (confirm that User A's tokens do *not* appear in the list)
- User A tries to `update` their own token so that it refers to User B as the owner (should fail with an appropriate error)